src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml.cpp

```diff
@@ -90,9 +90,25 @@ void EquationInputArea::InputTextBox_Submitted(Object ^ sender, RoutedEventArgs
     auto eq = static_cast<EquationViewModel ^>(tb->DataContext);
     eq->Expression = tb->GetEquationText();
 
-    if (tb->HasFocus)
+    if (eq->Expression != nullptr && eq->Expression->Length() > 0)
+    {
+        auto hasFocus = tb->HasFocus;
+        unsigned int index = 0;
+        if (Equations->IndexOf(eq, &index) && index == Equations->Size - 1)
+        {
+            AddNewEquation();
+        }
+        if (hasFocus)
+        {
+            FocusManager::TryMoveFocus(::FocusNavigationDirection::Left);
```

Comment:
I think this would feel better if we gave focus to the ghost equation box!

Reply:
Modified. Not only it gives the focus to the ghost equation box but I've added an extra feature:
- If you press enter on an Equation box, it will automatically give the focus to the next one (or create a new one if it's the last)